### PR TITLE
only create new recognizer when new commands are added

### PR DESCRIPTION
**Pull request by corinnaj**

### CODE REVIEW

**Comment by corinnaj** [comment]
Whenever I call SpeechIn.Listen() with the same commands twice, it will not work any more and I get the following error:

```
ArgumentException: At least one keyword must be specified.
Parameter name: keywords
UnityEngine.Windows.Speech.KeywordRecognizer..ctor (System.String[] keywords, UnityEngine.Windows.Speech.ConfidenceLevel minimumConfidence)
WindowsSpeechIn.StartListening (System.String[] commands) (at Assets/unity-dualpanto-framework/Assets/SpeechIOForUnity/Assets/Scripts/Speech Input Core Scripts/WindowsSpeechIn.cs:44)
```

I changed it now so that it will only create a new recognizer when new keywords are added.
I am still slightly unsure why we need to check which arguments were already given. Does the new recognizer keep track of the old parameters? What happens if we want to remove arguments?

(Sorry about the amount of changes, my auto formatter kicked in, the relevant part is line 51)

**Comment by JotaroS** [comment]
Hi, LGTM, but I would recommend _not_ to include any reformatting commits for PR. I woud recommend sending separate PR for this (this will makes much more easier to review the code).

File: Assets/Scripts/Speech Input Core Scripts/WindowsSpeechIn.cs [hunk 1]
```diff
@@ -3,25 +3,30 @@
 using UnityEngine;
 using UnityEngine.Windows.Speech;
 
-public class WindowsSpeechIn : VoiceCommandBase {
```
Comment (JotaroS) [inline Assets/Scripts/Speech Input Core Scripts/WindowsSpeechIn.cs]: Could you undo reformatting related changes if possible? (only relavant commit should be highlited)

Comment (corinnaj) [reply Assets/Scripts/Speech Input Core Scripts/WindowsSpeechIn.cs]: alright, I reverted the other changes. Would you be alright with me making a seperate commit or PR using the default VS Code C# formatting?

Comment (JotaroS) [reply Assets/Scripts/Speech Input Core Scripts/WindowsSpeechIn.cs]: That would be ok. Just make sure you title the PR saying "[formatting] bla bla"

**Pull request merged**